From df1c50b4f783536322e0bf773c7483f4520c572f Mon Sep 17 00:00:00 2001
From: Mohan Kumar G <quic_mkumarg@quicinc.com>
Date: Thu, 23 Nov 2023 20:43:57 +0530
Subject: [PATCH] hostapd: enable beacon protection as default for 11be

Currently, beacon protection is not enabled as default
in EHT mode.

Beacon protection requires 11be to be enabled and security to
be set for each bss.

Fix this issue by checking if 11be is enabled and ieee80211w is
set for each bss.

Signed-off-by: Mohan Kumar G <quic_mkumarg@quicinc.com>
---
 hostapd/config_file.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

--- a/hostapd/config_file.c
+++ b/hostapd/config_file.c
@@ -4903,8 +4903,14 @@ struct hostapd_config * hostapd_config_r
 
 	fclose(f);
 
-	for (i = 0; i < conf->num_bss; i++)
+	for (i = 0; i < conf->num_bss; i++) {
 		hostapd_set_security_params(conf->bss[i], 1);
+#ifdef CONFIG_IEEE80211BE
+		if (conf->ieee80211be && conf->bss[i]->ieee80211w > 0
+		    && !conf->bss[i]->disable_11be)
+			conf->bss[i]->beacon_prot = 1;
+#endif /* CONFIG_IEEE80211BE */
+	}
 
 	if (hostapd_config_check(conf, 1))
 		errors++;
